src/modules/variable/get.rs

```diff
@@ -23,10 +26,22 @@ impl VariableGet {
 
 impl Typed for VariableGet {
     fn get_type(&self) -> Type {
-        match (&*self.index, self.kind.clone()) {
-            // Return the type of the array element if indexed
-            (Some(_), Type::Array(kind)) => *kind,
-            _ => self.kind.clone()
+        if let Type::Array(item_kind) = &self.kind {
+            if let Some(index) = self.index.as_ref() {
+                if let Some(ExprType::Range(_)) = &index.value {
+                    // Array type (indexing array by range)
+                    self.kind.clone()
+                } else {
+                    // Item type (indexing array by number)
+                    *item_kind.clone()
+                }
+            } else {
+                // Array type (returning array)
+                self.kind.clone()
+            }
+        } else {
+            // Variable type (returning text or number)
+            self.kind.clone()
         }
```

Comment:
Lets use a match expression testing for `(&self.kind, self.index.as_ref())` so you can have the `(Type::Array(item_kind), Some(index))` match arm for you logic, and the default `_` match arm to remove the duplication of `self.kind.clone()`.

Reply:
Changed to use match guards, but removal of duplication does not make sense, given the way I did it.